plan/heuristic_planner.go

```diff
@@ -21,6 +21,20 @@ func (p *heuristicPlanner) addRules(rules ...Rule) {
 	}
 }
 
+func (p *heuristicPlanner) removeRule(ruleName string) bool {
+	var removed bool
+	for _, rulesPerKind := range p.rules {
```

Comment:
Don't you think it would have the same complexity?

If working on a batch of rules (i.e. `removeRules(ruleNames ...string)`) I should nest another `for` to iterate over `ruleNames`. The complexity remains the same if somebody invokes `removeRule` for each rule to be removed.

Reply:
But what if you make the ruleNames a HashSet?

Reply:
Yep, I see your point. I'll update this 👍